ui/mirage/factories/agent.js

```diff
@@ -8,20 +8,40 @@ const AGENT_STATUSES = ['alive', 'leaving', 'left', 'failed'];
 
 export default Factory.extend({
   id: i => (i / 100 >= 1 ? `${UUIDS[i]}-${i}` : UUIDS[i]),
-  name: () => `nomad@${faker.random.boolean() ? faker.internet.ip() : faker.internet.ipv6()}`,
 
-  status: () => faker.helpers.randomize(AGENT_STATUSES),
-  serfPort: () => faker.random.number({ min: 4000, max: 4999 }),
-
-  address() {
-    return this.name.split('@')[1];
+  config: {
+    Version: {
+      Version: '1.1.0',
+      VersionMetadata: 'ent',
+      VersionPrerelease: 'dev',
+    },
   },
 
-  tags() {
-    const rpcPortCandidate = faker.random.number({ min: 4000, max: 4999 });
+  member: () => {
+    const name = generateName();
+    const serfPort = faker.random.number({ min: 4000, max: 4999 });
     return {
-      port: rpcPortCandidate === this.serfPort ? rpcPortCandidate + 1 : rpcPortCandidate,
-      dc: faker.helpers.randomize(DATACENTERS),
+      Name: name,
+      SerfPort: serfPort,
```

Comment:
Just this be just `Port` to match the API result?

```json
  "member": {
    "Addr": "172.31.14.218",
    "DelegateCur": 4,
    "DelegateMax": 5,
    "DelegateMin": 2,
    "Name": "ip-172-31-14-218.global",
    "Port": 4648,
    "ProtocolCur": 2,
    "ProtocolMax": 5,
    "ProtocolMin": 1,
    "Status": "alive",
    "Tags": {
      "dc": "dc1",
      "mvn": "1",
      "bootstrap": "1",
      "id": "a9a68da9-0d4e-c9fa-6774-4017cfdae40d",
      "port": "4647",
      "expect": "1",
      "rpc_addr": "172.31.14.218",
      "role": "nomad",
      "region": "global",
      "vsn": "1",
      "build": "0.12.12",
      "raft_vsn": "2"
    }
  },
```

Reply:
Great callout.